packages/transformer/src/IModelTransformer.ts

```diff
@@ -2231,6 +2228,31 @@ export class IModelTransformer extends IModelExportHandler {
       (this.targetDb as any).codeValueBehavior = "trim-unicode-whitespace";
     }
     /* eslint-enable @itwin/no-internal */
+
+    const defaultSaveTargetChanges = () => this.targetDb.saveChanges();
+    await (saveTargetChanges ?? defaultSaveTargetChanges)(this);
+    if (this.isReverseSynchronization) this.sourceDb.saveChanges();
+
+    const description = `${
+      this._isProvenanceInitTransform
+        ? `initialized branch provenance with master`
+        : this.isForwardSynchronization
+          ? "Forward sync of"
+          : "Reverse sync of"
+    } iModel: ${this.sourceDb.iModelId}`;
+
+    if (this.targetDb.isBriefcaseDb()) {
+      // FIXME<NICK> This relies on them defining an authorizationClient on iModelHost, otherwise this will fail. Not sure thats good.
+      await this.targetDb.pushChanges({
+        description,
+      });
+    }
```

Comment:
I would rather this be overriddable. What's the reason for overriding `saveTargetChanges` anyway? If necessary, both could be overriddable.

Reply:
it stinks that we can't locally save/queue a changeset without immediately pushing it...
I would much prefer that we separated the changesets while allowing the user to choose when they push...

Reply:
Not sure about saveTargetChanges like we mentioned in private chat, that was already present.

I add support for custom changeset descriptions. I will add tests for it into the PR at the top of the stack since I think there will be some conflicts if I do it here.